controllers/v1/programs.js

```diff
@@ -0,0 +1,113 @@
+const config = require('../../config/config');
+const rp = require('request-promise');
+const request = require('request');
+const model = require('../../db');
+const helperFunc = require('../../helper/chart_data');
+const authService = require('../../middleware/authentication_service');
+
+
+/**
+   * @api {post} /dhiti/api/v1/programs/list 
+   * Programs List 
+   * @apiVersion 1.0.0
+   * @apiGroup Programs
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+  "entityId": "",
+  "entityType": ""
+* }
+   * @apiSuccessExample {json} Success-Response:
+*     HTTP/1.1 200 OK
+*     {
+       "result": true,
+       "data": [
+           {
+               "programId": "",
+               "programName": "",
+               "type": ""
+           }
+       ]
+*     }
+   * @apiUse errorBody
+   */
+
+exports.list = async function (req, res) {
+  
+    if (!req.body.entityId || !req.body.entityType) {
+        res.status(400);
+        var response = {
+            result: false,
+            message: 'entityId and entityType are required fields'
+        }
+        res.send(response);
+    }
+    else {
+         
+         //get quey from cassandra
+         model.MyModel.findOneAsync({ qid: "programs_list_query" }, { allow_filtering: true })
+         .then(async function (result) {
+
+             let bodyParam = JSON.parse(result.query);
+
+             let id = await getCreatedByField(req,res);
+
+             bodyParam.filter.fields[0].dimension = req.body.entityType;
+             bodyParam.filter.fields[0].value = req.body.entityId;
+
+             //fetch assessment programs from assessment datasource
+             if (config.druid.assessment_datasource_name) {
+                 bodyParam.dataSource = config.druid.assessment_datasource_name;
+             }
+             
+             bodyParam.filter.fields[1].fields[0].fields[0].dimension = "userId";
+             bodyParam.filter.fields[1].fields[0].fields[0].value = id;
+
+             let options = config.druid.options;
+             options.method = "POST";
+             options.body = bodyParam;
+
+             let assessmentPrograms = await rp(options);
```

Comment:
@deepakr123 - Run queries in both data sources in parallel

Reply:
resolved